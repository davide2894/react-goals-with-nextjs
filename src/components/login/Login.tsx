import { useState, SyntheticEvent, ChangeEvent } from "react";
import { loginWithEmailAndPassword } from "@firebase";
import { AuthAction, withAuthUser } from "next-firebase-auth";
import Loader from "@components/loader/Loader";
import SubmitButton from "@components/submitButton/SubmitButton";

function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  function handleSubmit(evt: SyntheticEvent) {
    evt.preventDefault();
    loginWithEmailAndPassword(email, password);
  }

  function handleEmailChange(evt: ChangeEvent<HTMLInputElement>) {
    setEmail(evt.target.value);
  }

  function handlePasswordChange(evt: ChangeEvent<HTMLInputElement>) {
    setPassword(evt.target.value);
  }

  return (
    <div
      data-testid="loginComponentTest"
      className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4">
      <h2 className="text-gray-700 font-bold">
        Login if you have already an account
      </h2>
      <br></br>
      <form onSubmit={handleSubmit}>
        <div className="mb-2">
          <label
            className="text-gray-700 text-sm font-bold mb-2"
            htmlFor="loginFormEmailInput">
            Email
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            type="email"
            name="email"
            required
            id="loginFormEmailInput"
            value={email}
            onChange={(evt) => handleEmailChange(evt)}
          />
          <span className="separator"> </span>
        </div>
        <div className="mb-4">
          <label
            className="text-gray-700 text-sm font-bold mb-2"
            htmlFor="loginFormPassword">
            Password
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            type="password"
            name="password"
            required
            id="loginFormPassword"
            value={password}
            onChange={(evt) => handlePasswordChange(evt)}
          />
          <span className="separator"> </span>
        </div>
        <div>
          <SubmitButton text="Login" />
        </div>
      </form>
    </div>
  );
}

export default withAuthUser({
  whenAuthed: AuthAction.REDIRECT_TO_APP,
  whenUnauthedBeforeInit: AuthAction.SHOW_LOADER,
  whenUnauthedAfterInit: AuthAction.RENDER,
  LoaderComponent: Loader,
})(Login);

// export const getServerSideProps = withAuthUserSSR({
//   whenUnauthed: AuthAction.REDIRECT_TO_LOGIN,
// })();
