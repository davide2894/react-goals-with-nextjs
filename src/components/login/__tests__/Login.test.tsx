import Login from "../Login";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Login component renders correctly", () => {
  render(<Login />);
  const loginComponentWrapper = screen.getByTestId("loginComponentTest");
  expect(loginComponentWrapper).toBeInTheDocument();
  expect(loginComponentWrapper).toHaveClass("bg-white");
  expect(loginComponentWrapper).toHaveClass("shadow-md");
});

test("Login component matches its own snapshot", () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
