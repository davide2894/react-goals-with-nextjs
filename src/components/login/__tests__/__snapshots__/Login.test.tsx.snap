// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches its own snapshot 1`] = `
<div
  className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4 w-[380px] h-[380px]"
  data-testid="loginComponentTest"
>
  <h2
    className="text-gray-700 font-bold mb-6"
  >
    Login if you have already an account
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="mb-2"
    >
      <label
        className="text-gray-700 text-sm font-bold mb-2"
        htmlFor="loginFormEmailInput"
      >
        Email
      </label>
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="loginFormEmailInput"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="text-gray-700 text-sm font-bold mb-2"
        htmlFor="loginFormPassword"
      >
        Password
      </label>
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="loginFormPassword"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div>
      <button
        className="bg-yellow-500 hover:bg-yellow-700 text-black font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
        type="submit"
        value="submit"
      >
        Login
      </button>
    </div>
  </form>
</div>
`;
